### ISSUE TITLE
urllib3 dependency is missing

### ISSUE THREAD

**Issue by codingcatgirl** [alanhamlett/pip-update-requirements#47]
What it says in the title. If i install pur in python3.10 using pip, i can't run it because it's missing urllib3.

`pip install urllib3` fixes the issue.

**Comment by alanhamlett** [comment]
That's strange because pip is the one needing urllib3 and it comes vendored with it:

https://github.com/alanhamlett/pip-update-requirements/tree/c4dbf6c5c31b959a6c3556848fe6bae9c89c41ca/pur/packages/pip/_vendor/urllib3

Most likely it's accidentally doing `import urllib3` instead of `import pip._vendored.urllib3` somewhere.

**Comment by alanhamlett** [comment]
@codingcatgirl what's the full traceback shown when trying to run `pur`?

**Comment by codingcatgirl** [comment]
```
/tmp/ pur
Traceback (most recent call last):
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 27, in <module>
    from . import urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/__init__.py", line 8, in <module>
    from .connectionpool import (
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 42, in <module>
    from .response import HTTPResponse
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/response.py", line 8, in <module>
    from ._collections import HTTPHeaderDict
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/_collections.py", line 2, in <module>
    from collections import Mapping, MutableMapping
ImportError: cannot import name 'Mapping' from 'collections' (/usr/lib/python3.10/collections/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/env/lib/python3.10/site-packages/pur/__init__.py", line 27, in <module>
    from pip._internal.network.session import PipSession
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_internal/network/session.py", line 20, in <module>
    from pip._vendor import requests, urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/__init__.py", line 136, in <module>
    from . import packages
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 29, in <module>
    import urllib3
ModuleNotFoundError: No module named 'urllib3'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 27, in <module>
    from . import urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/__init__.py", line 8, in <module>
    from .connectionpool import (
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 42, in <module>
    from .response import HTTPResponse
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/response.py", line 8, in <module>
    from ._collections import HTTPHeaderDict
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/_collections.py", line 2, in <module>
    from collections import Mapping, MutableMapping
ImportError: cannot import name 'Mapping' from 'collections' (/usr/lib/python3.10/collections/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/env/bin/pur", line 5, in <module>
    from pur.__init__ import pur
  File "/tmp/env/lib/python3.10/site-packages/pur/__init__.py", line 35, in <module>
    from pip._internal.network.session import PipSession
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_internal/network/session.py", line 20, in <module>
    from pip._vendor import requests, urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/__init__.py", line 136, in <module>
    from . import packages
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 29, in <module>
    import urllib3
ModuleNotFoundError: No module named 'urllib3'
```

**Comment by codingcatgirl** [comment]
It looks a bit like it's a bug in pip… but pip works just fine. Maybe you're just not meant to import from _vendored? That's a private module after all.

**Comment by alanhamlett** [comment]
What's your Python version? `python --version` or `python3 --version`

I've tested on Python 3.10.2 and tests are passing and I can run `pur --help`.

Does this work if you run it with Python 3.10? `python -c "from collections.abc import Mapping"`

Maybe your python installation is corrupted and somehow missing the `collections.abc` module.

**Comment by codingcatgirl** [comment]
It's python 3.10.4. It can't be that this module is missing because installing urllib3 solves it. Also yes, i can import it.

It might be relevant to note that i'm running this in a virtual environment.

Here's the full steps to reproduce for me.

```
█▓▒░laura@desktop█▓▒░ Mi Apr 06 12:39:24 
/tmp/ python --version
Python 3.10.4

█▓▒░laura@desktop█▓▒░ Mi Apr 06 12:39:33 
/tmp/ python -m venv env

█▓▒░laura@desktop█▓▒░ Mi Apr 06 12:39:37 
/tmp/ . env/bin/activate

█▓▒░laura@desktop█▓▒░ Mi Apr 06 12:39:43 
/tmp/ pip --version
pip 22.0.4 from /tmp/env/lib/python3.10/site-packages/pip (python 3.10)

(env) █▓▒░laura@desktop█▓▒░ Mi Apr 06 12:39:47 
/tmp/ pip install pur
Collecting pur
  Using cached pur-6.0.1-py3-none-any.whl (2.5 MB)
Collecting click>=8.0.0
  Using cached click-8.1.2-py3-none-any.whl (96 kB)
Installing collected packages: click, pur
Successfully installed click-8.1.2 pur-6.0.1

(env) █▓▒░laura@desktop█▓▒░ Mi Apr 06 12:39:58 
/tmp/ pur
Traceback (most recent call last):
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 27, in <module>
    from . import urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/__init__.py", line 8, in <module>
    from .connectionpool import (
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 42, in <module>
    from .response import HTTPResponse
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/response.py", line 8, in <module>
    from ._collections import HTTPHeaderDict
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/_collections.py", line 2, in <module>
    from collections import Mapping, MutableMapping
ImportError: cannot import name 'Mapping' from 'collections' (/usr/lib/python3.10/collections/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/env/lib/python3.10/site-packages/pur/__init__.py", line 27, in <module>
    from pip._internal.network.session import PipSession
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_internal/network/session.py", line 20, in <module>
    from pip._vendor import requests, urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/__init__.py", line 136, in <module>
    from . import packages
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 29, in <module>
    import urllib3
ModuleNotFoundError: No module named 'urllib3'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 27, in <module>
    from . import urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/__init__.py", line 8, in <module>
    from .connectionpool import (
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 42, in <module>
    from .response import HTTPResponse
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/response.py", line 8, in <module>
    from ._collections import HTTPHeaderDict
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/_collections.py", line 2, in <module>
    from collections import Mapping, MutableMapping
ImportError: cannot import name 'Mapping' from 'collections' (/usr/lib/python3.10/collections/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/env/bin/pur", line 5, in <module>
    from pur.__init__ import pur
  File "/tmp/env/lib/python3.10/site-packages/pur/__init__.py", line 35, in <module>
    from pip._internal.network.session import PipSession
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_internal/network/session.py", line 20, in <module>
    from pip._vendor import requests, urllib3
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/__init__.py", line 136, in <module>
    from . import packages
  File "/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/__init__.py", line 29, in <module>
    import urllib3
ModuleNotFoundError: No module named 'urllib3'

(env) █▓▒░laura@desktop█▓▒░ Mi Apr 06 12:40:03 
/tmp/ python
Python 3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Mapping
>>> # no error
>>> 
>>> 

(env) █▓▒░laura@desktop█▓▒░ Mi Apr 06 12:40:30 
/tmp/ pip install urllib3
Collecting urllib3
  Using cached urllib3-1.26.9-py2.py3-none-any.whl (138 kB)
Installing collected packages: urllib3
Successfully installed urllib3-1.26.9

(env) █▓▒░laura@desktop█▓▒░ Mi Apr 06 12:40:38 
/tmp/ pur 
Error: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'
```

**Comment by alanhamlett** [comment]
I'm able to reproduce it when installing `pur` in a virtualenv. When I look at `/tmp/env/lib/python3.10/site-packages/pur/packages/pip/_vendor/requests/packages/urllib3/_collections.py`, it's contents are somehow from [an old version back in 2016](https://github.com/pypa/pip/blob/aab40c24e55fdf633f98229c90fc582469e9d98e/pip/_vendor/requests/packages/urllib3/_collections.py).

I've released a new version 6.1.0 from a fresh build. That fixed it for me, you too? `pip install --upgrade pur && pip uninstall urllib3`.

**Comment by codingcatgirl** [comment]
Yes, that fixed it!
